src/main/scala/net/snowflake/spark/snowflake/SnowflakeRelation.scala

```diff
@@ -130,7 +130,7 @@ private[snowflake] case class SnowflakeRelation(
             sqlContext.getConf("spark.sql.shuffle.partitions", "200").toInt
           val emptyRow = Row.empty
           sqlContext.sparkContext
-            .parallelize(1L to numRows, parallelism)
+            .range(start = 0L, end = numRows, numSlices = parallelism)
```

Comment:
For `DataFrame.count()`, if it is pushdown to snowflake, the final return be a single Long value.
I don't understand why this change can affect the issue.

Reply:
@sfc-gh-mrui Thanks for your reply.
Yes, and here is building RDD with `result` number of empty rows, but `SparkContext.parallelize()` will fail to do it, for its first parameter `Seq[T]`'s `length()` only supports Int32

Reply:
Query wouldn't be pushed down if spark version is lower than 3.1 and `requiredColumns` of `buildScan()` is empty, instead, it will get here.

Reply:
I think the `count()` can be pushdown for  spark 3.1.
Please make sure you select the corresponding spark connector version for the spark version.
For example, for spark connector 2.9.1, we released binaries for spark 2.4, 3.0 and 3.1.

Reply:
Right, we are using Spark 3.0.2 and spark connector 2.9.1

Reply:
https://mvnrepository.com/artifact/net.snowflake/spark-snowflake_2.12/2.9.1-spark_3.0

Reply:
@dohongdayi After explicitly disable pushdown for the test case, I can reproduce the issue. And I can ack this fix can fix that problem. After you update the test case. I can approve this PR.